docs/user/openstack/ovs-dpdk.md

```diff
@@ -0,0 +1,244 @@
+a virtualized IOMMU and we need to explicitly enable it via `options vfio enable_unsafe_noiommu_mode=1`.
+This can be done via a MachineConfig resource:
+
+``` yaml
+kind: MachineConfig
+apiVersion: machineconfiguration.openshift.io/v1
+metadata:
+  name: 99-vfio-noiommu 
+  labels:
+    machineconfiguration.openshift.io/role: worker
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+      - path: /etc/modprobe.d/vfio-noiommu.conf
+        mode: 0644
+        contents:
+          source: data:;base64,b3B0aW9ucyB2ZmlvIGVuYWJsZV91bnNhZmVfbm9pb21tdV9tb2RlPTEK
+```
+
+More documentation about that can be found [here][3].
+
+### Bind the `vfio-pci` kernel driver to the appropriate NICs
+
+The workers connected to the `VFIO_NETWORK` require the `vfi-pci` kernel driver to be bound to the ports attached to the `VFIO_NETWORK`.
+
+We basically need to create a MachineSet for the workers which will be attached to the `VFIO_NETWORK` network.
+The MachineConfig used in the MachineSet should contain the necessary SystemD unit definitions to install the poll mode driver (`vfio-pci`) on boot.
+This [Ansible role][4] provides an easy way to generate the required MachineSet.
+The playbooks also provide the ability to automatically install the MachineSet, but it is highly recommended to use the playbooks for generating a MachineSet file (.yaml).
+Then the user should explicitly install this on their cluster using the appropriate `oc` commands.
+
+To generate the MachineSet file, run the following commands:
+
+``` bash
+openstack network show <name of VFIO network> -f value -c id
+ansible-playbook play.yaml -e network_ids="ID of VFIO network" --extra-vars "mc_config_file=/tmp/vhostuser.yaml"
+oc create -f /tmp/vhostuser.yaml
+```
+
+The MachineSet will enable a script at boot which will inspects the config drive attached to the VM for the required information. If not found there, it queries the metadata server for it. Based on the network ID, the MAC address of the port is learned and this is translated into the correct PCI bus ID. This means that on boot the vfio-pci module is bound to any port that is attached to the network identified by the OpenStack network ID.
+
+The result is that the interface will have the vfio-pci driver bound to it. You can verify this by executing the `lspci -k` command on the workers in question. The result should be similar to this:
+
+``` bash
+lspci -k
+...
+00:07.0 Ethernet controller: Red Hat, Inc. Virtio network device
+	Subsystem: Red Hat, Inc. Device 0001
+	Kernel driver in use: vfio-pci
+```
+
+As you can see, the Ethernet controller at bus ID `00:07.0` is using the vfio-pci kernel driver.
+
+### Expose host-device interface to pod
+
+The host-device CNI plugin can be used to expose an interface on the host to the pod.
+The plugin moves the interface from the host network namespace to the pods namespace.
+Effectively given direct control of the interface to the pod running in the namespace.
+
```

Comment:
Namespace `example-cnf` should be created

Reply:
I propose that we document it like: `namespace: <your-cnf-namespace>`.
Users know how to create a namespace and if they don't, they should look at the OpenShift doc, but not here. Trying to keep the content short and concise.